panther_docking/src/dock_pose_publisher_node.cpp

```diff
@@ -23,81 +25,100 @@
 #include <geometry_msgs/msg/pose_stamped.hpp>
 #include <tf2_geometry_msgs/tf2_geometry_msgs.hpp>
 
-class DockPosePublisherNode : public rclcpp::Node
+namespace panther_docking
+{
+DockPosePublisherNode::DockPosePublisherNode(const std::string & name) : Node(name)
 {
-public:
-  DockPosePublisherNode() : Node("pose_publisher_node")
-  {
-    declare_parameter("publish_rate", 10.0);
-    declare_parameter("docks", std::vector<std::string>({"main"}));
-    declare_parameter("fixed_frame", "odom");
-
-    const auto fixed_frame = get_parameter("fixed_frame").as_string();
-    const auto docks = get_parameter("docks").as_string_array();
-
-    std::string dock_pose_frame_id;
-    for (const auto & dock : docks) {
-      declare_parameter(dock + ".type", "panther_charging_dock");
-      declare_parameter(dock + ".frame", "main_wibotic_receiver_requested_pose_link");
-
-      const auto dock_type = get_parameter(dock + ".type").as_string();
-      if (dock_type == "panther_charging_dock") {
-        dock_pose_frame_id = get_parameter(dock + ".frame").as_string();
-      }
+  declare_parameter("publish_rate", 10.0);
+  declare_parameter("docks", std::vector<std::string>({"main"}));
```

Comment:
I wonder if there is any point in creating this node. It acts as a data converter, which is more like a helper function for panther_utils. than a sensible node. If you have a different opinion, I think we can ask someone else for an opinion.

Reply:
This is a data converter from tfs to Pose message. 
In the first version of my docking plugin, tfs were read directly in the plugin but it didn't work on the real robot because if the latency. 

The subscribers work "asynchronously" form the plugin's point of view.

This node cannot be moved to `panther_utils` because it is strictly connected to parameters of `docking_server`  defined in the `panther_docking_server.yaml`

Reply:
I will think about this.